src/client/components/HealthCareApp.jsx

```diff
@@ -84,17 +85,14 @@ class HealthCareApp extends React.Component {
 
   handleSubmit(e) {
     e.preventDefault();
-    const path = this.props.location.pathname;
-    const store = this.context.store;
-    const veteran = store.getState().veteran;
+    const veteran = this.props.data;
 
     // Strip out unnecessary fields that track UI state
     function reducer(i, d) {
       return typeof d.value !== 'undefined' ? d.value : d;
     }
 
-    store.dispatch(updateSubmissionStatus('submitPending'));
-    store.dispatch(updateCompletedStatus(path));
+    this.props.onUpdateSubmissionStatus('submitPending');
 
     // POST data to endpoint
```

Comment:
Let's first check if there are any validation errors in the form, and then only post data conditionally on that being false.